File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/actions/PivotUpAction.java
package org.firstinspires.ftc.teamcode.actions;

import com.acmerobotics.dashboard.telemetry.TelemetryPacket;
import com.acmerobotics.roadrunner.Action;
import com.qualcomm.robotcore.hardware.DcMotor;
import com.qualcomm.robotcore.hardware.HardwareMap;

public class PivotUpAction implements Action {
    private final DcMotor pivot;
    private Long started;

    public PivotUpAction(HardwareMap hardwareMap) {
        this.pivot = hardwareMap.dcMotor.get("pivot");
        this.started = null;
    }

    @Override
    public boolean run(TelemetryPacket p) {
        if (started == null) {
            started = System.currentTimeMillis();
            pivot.setPower(0.8);
            return true;
        }

        return System.currentTimeMillis() - started <= 1300;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Actions/PivotUpAction.kt
package org.firstinspires.ftc.teamcode.Actions

import com.acmerobotics.dashboard.telemetry.TelemetryPacket
import com.acmerobotics.roadrunner.Action
import com.qualcomm.robotcore.hardware.HardwareMap

class PivotUpAction(private val hardwareMap: HardwareMap) : Action {
    private val pivot = hardwareMap.dcMotor.get("pivot")
    private var started: Long? = null

    override fun run(p: TelemetryPacket): Boolean {
        if (started == null) {
            started = System.currentTimeMillis()
            pivot.power = 0.8
            return true
        }

        return System.currentTimeMillis() - started!! <= 1300
    }


}